File: Valid_Paren.py
class Solution:
    def isValid(self, s: str) -> bool:
        bDict = {'{':'}','[':']','(':')'}
        stack = [];
        if len(s)==0: 
            return True
        if len(s)%2!=0:
            return False
        for letter in s:
            if letter in bDict.keys():
                stack.append(letter)
            else:
                if len(stack) == 0:
                    return False
                elif letter!=bDict[stack.pop(-1)]:
                    return False
        if len(stack)>0:
            return False
        return True
                
            
        

File: Valid_Paren.js
var isValid = function(s) {
    const bDict = {'{':'}', "[":"]", "(":")"};
    const stack = [];
    const sLen = s.length;
    if(sLen===0) return true;
    if(sLen%2!==0) return false;
    for (let i = 0; i<sLen; i++){
        if(s[i] in bDict){
            stack.push(s[i]);
        }
        else{
            if(i<1) return false;
            if(bDict[stack.pop()]!=s[i]) return false;
        }
    }
    if(stack.length!=0) return false;
    return true;
};
